### ISSUE TITLE
Set Fallback charset setting to UTF-8 to avoid mojibake (garbled text)

### ISSUE THREAD

**Issue by kolAflash** [RealRaven2000/SmartTemplate4#338]
I'm irritated by the `charset` setting in the `Global Settings`. And I couldn't find documentation about it.

I know quite well what a charset is, being a software developer myself. But I'm unsure what the setting in SmartTemplate actually does.

&nbsp;

Does this setting change the mail encoding when writing, replying or forwarding mails?  
So why not use Thunderbird's default encoding setting?

After installing SmartTemplate on two computers running Thunderbird-128 on Debian-12 (Linux, KDE), the setting defaulted to `ISO-8859-1`. That looks pretty antique to me. Shouldn't mails be UTF-8 encoded nowadays?

And there's no indication if SmartTemplate accepts the encoding value I entered. Is it `UTF8` or `UTF-8` for example?  
Additional to an validity indicator, a short description on how possible values look like would be helpful.

And beside values for actual encodings like `UTF-8` or `ISO-8859-1` it would be helpful, if one could simply enter `thunderbird-default`, which would then use Thunderbird's default encoding.  
_(as far as I understand that setting, this would be my preferred value for it)_

&nbsp;

Thanks for the great Open Source work!  
kolAflash  
_(an otherwise happy and paying user)_

**Comment by RealRaven2000** [comment]
> Does this setting change the mail encoding when writing, replying or forwarding mails?
> So why not use Thunderbird's default encoding setting?

This is a legacy setting that has been in the Add-no forever, IIRC, it's not actually about the charset it uses itself but about data that it reads from the original email (the one you reply to). It is also actually a fallback when SmartTemplates tries to decode things like subject lines and headers that have encoded characters in them.

The function that uses the setting is [`SmartTemplate4.mimeDecoder.detectCharset()`](https://github.com/RealRaven2000/SmartTemplates/blob/ESR115/chrome/content/smartTemplate-overlay.js#L594), which is itself used by the `SmartTemplate4.mimeDecoder.decode()` (used for decoding mail headers) and `SmartTemplate4.mimeDecoder.split()` (which reads address headers) functions - which is used to , as this is currently not using API functions to read headers it didn't have the luxury of already decoded header fields.

Not quite sure whether this settings should be completely hidden or whether there is a safe way of rewriting all that code without breaking more esoteric / old / non-standard mailing systems...

**Comment by kolAflash** [comment]
I think documenting what it does already helps a lot.  
Thanks!

&nbsp;

**So do I understand this correctly?**
- It's about decoding existing mails, not about creating mails.
- And it only gets used if no charset can be automatically detected.
  - So in that case the default value `ISO-8859-1` would make it assume, that an email with unclear charset is ISO-8859-1. And that's probably a good guess for old mails without charset information.

**Comment by RealRaven2000** [comment]
> **So do I understand this correctly?**
> 
>     * It's about decoding existing mails, not about creating mails.
> 
>     * And it only gets used if no charset can be automatically detected.
>       
>       * So in that case the default value `ISO-8859-1` would make it assume, that an email with unclear charset is ISO-8859-1. And that's probably a good guess for old mails without charset information.

yes, I think that's pretty much correct. It often affects things like names in email addresses or subject lines, e.g. from Russian / eastern Asisan users.

**Comment by RealRaven2000** [comment]
I am switching the default charset to "UTF-8" to avoid mojibake, and am also providing a better label and a tooltip button:

![Image](https://github.com/user-attachments/assets/6c33c241-b570-4f59-b129-6c3bdfd14945)

I have been working on the tooltip text code to be reusable so I can add more tooltip button to other elements in the settings pages.

**Comment by RealRaven2000** [comment]
Here is a version with the new default and a new tooltip system which explains the usage better:

[smartTemplate-fx-4.10.2pre72.zip](https://github.com/user-attachments/files/18933470/smartTemplate-fx-4.10.2pre72.zip)

----

To try out this version, download the zip file and then drag it into Thunderbird Add-ons Manager (without extracting)

**Comment by kolAflash** [comment]
> Here is a version with the new default and a new tooltip system which explains the usage better:
> [...]

Perfect 👍 😃

**Comment by RealRaven2000** [comment]
Yeah I wanna add more usability features in the settings dialog - currently I am working on [some issues](https://github.com/RealRaven2000/SmartTemplates/issues/359) caused by the JAWS screen reader. I will close this issue myself once I release the new SmartTemplates for the users, so please leave it open until then.

I hope I can make the new tooltips easily accessibly to screenreaders as well, as the SmartTemplates settings dialog is somewhat of a maze, even to seeing eyes. Tooltips seems a bit of an antipattern, so I am going to slowly convert these to something more useful.

**Comment by MarkyMarkDE** [comment]
4.1.1pre154: The tooltip content is not readable with dark skin and effects all info-boxes which use the same method, follow is an example only:

![Image](https://github.com/user-attachments/assets/0e59ce3a-03eb-4c9b-b4be-6c97406d4874)

**Comment by RealRaven2000** [comment]
> The tooltip content is not readable with dark skin and effects all info-boxes which use the same method

Good catch! Here is a new version:

[smartTemplate-fx-4.11pre164.zip](https://github.com/user-attachments/files/18974167/smartTemplate-fx-4.11pre164.zip)

I am currently working on [accessibility improvements for screenreaders](https://github.com/RealRaven2000/SmartTemplates/issues/361), so there maybe some fallout with the layout, please keep a lookout for other such weirdness!

The tooltip buttons will probably get a slightly different icon as well (maybe just a white question mark on dark / blue circle) to distinguish them from the links to the documentation, which currently use the same one.

**Comment by MarkyMarkDE** [comment]
good job, thank you!
Now it matches my dark skin really well and is readable.
If possible but not important: try to give the box a 5px border-radius so it looks more softly, so we need also maybe little margins around the texts, so they will not be cutted off :wink:

![Image](https://github.com/user-attachments/assets/fd966ba0-f4eb-4b3c-b7d2-83edf5f26efc)

New icon: you know, its okay with me, do whatever you need to do :wink:
Sure I will keep my eyes stays open.

**Comment by RealRaven2000** [comment]
Implemented in 4.11 - Published 05/04/2025
